cli/pack/docker.go

```diff
@@ -0,0 +1,110 @@
+package pack
+
+import (
+	_ "embed"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/otiai10/copy"
+	"github.com/tarantool/tt/cli/cmdcontext"
+	"github.com/tarantool/tt/cli/config"
+	"github.com/tarantool/tt/cli/docker"
+	"github.com/tarantool/tt/cli/templates/engines"
+)
+
+//go:embed templates/Dockerfile.pack.build
+var buildDockerfile []byte
+
+// RunInDocker runs tt pack in docker container.
+func RunInDocker(cmdCtx *cmdcontext.CmdCtx, opts config.CliOpts, cmdArgs []string) error {
+	tmpDir, err := ioutil.TempDir("", "docker_pack_ctx")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	envDir := filepath.Dir(cmdCtx.Cli.ConfigPath)
+
+	goTextEngine := engines.NewDefaultEngine()
+	dockerfileText, err := goTextEngine.RenderText(string(buildDockerfile),
+		map[string]string{
+			"tnt_version": cmdCtx.Cli.TarantoolVersion,
+			"env_dir":     filepath.Base(envDir),
+		})
+	if err != nil {
+		return err
+	}
+
+	// Write docker file (rw-rw-r-- permissions).
+	if err = ioutil.WriteFile(filepath.Join(tmpDir, "Dockerfile"),
+		[]byte(dockerfileText),
+		0664); err != nil {
+		return err
+	}
+
+	// Remove --use-docker from args.
+	for i, arg := range cmdArgs {
+		if arg == "--use-docker" {
+			cmdArgs = append(cmdArgs[:i], cmdArgs[i+1:]...)
+		}
+	}
+
+	// Generate pack command line for tt in container.
+	ttPackCommandLine := append([]string{"tt", "pack"}, cmdArgs...)
+
+	// If bin_dir is not empty, we need to pack binaries built in container.
+	if _, err = os.Stat(opts.App.BinDir); err == nil {
+		relEnvBinPath, _ := filepath.Rel(envDir, opts.App.BinDir)
+		ttPackCommandLine = append([]string{"/bin/bash", "-c",
+			fmt.Sprintf("cp $(which tarantool) %s && cp $(which tt) %s && %s",
```

Comment:
`which tarantool` is not current environment tarantool. The same is applicable to tt also.

Reply:
That call is not for your host's environment. tt and tarntool are already installed in container and I use them for packaing.